src/generators/default_map_generator_job.cpp

```diff
@@ -1089,8 +1068,8 @@ std::string default_map_generator_job::default_generate_map(size_t width, size_t
 		// Search a path out for the road
 		pathfind::plain_route rt = pathfind::a_star_search(src, dst, 10000.0, &calc, width, height);
 
-		std::string road_base_name;
-		const std::string& name = generate_name(name_generator, "road_name", &road_base_name);
+		std::string road_base_name = base_name_generator->generate();
+		const std::string& name = road_name_generator->generate(std::map<std::string,std::string>{{{"base", road_base_name}}});
```

Comment:
Does this work without the cast? If not, it may be nicer to use a typedef. (I think there already is one for this particular type somewhere - `string_map`.)